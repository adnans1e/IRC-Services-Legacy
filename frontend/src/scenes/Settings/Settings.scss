.Settings {
	.theme {
		.active {
			font-weight: bold;
		}
	}
}