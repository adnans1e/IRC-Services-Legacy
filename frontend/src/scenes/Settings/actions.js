import { createAction } from 'redux-actions';

const actionTypes = {
	setLogoutTimeout: 'SET_LOGOUT_TIMEOUT',
}
export default {
	types: actionTypes,
	setLogoutTimeout: createAction(actionTypes.setLogoutTimeout),
};
