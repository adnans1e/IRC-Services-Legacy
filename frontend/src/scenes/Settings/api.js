import composeHeader from '../../data/Helpers/request';
import store from '../../shared/store';

const url = '/api/settings/';
let api = api || {};

api.settings = {
	getAll: () => {
		let {login} = store.getState();

		if(!login.user) return [];

		let headers = composeHeader(login.csrfToken, login.user.token);

		return fetch(`${url}`, {headers}).then(r => r.json());
	}
};

export default api;
