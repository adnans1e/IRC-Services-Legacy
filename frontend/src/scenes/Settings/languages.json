{
	"en": {
		"Settings": {
			"title": "Settings",
			"description": "Provider Types Description",
			"properties": {
				"width": "long",
				"height": "not long"
			}
		}
	},
	"fr": {
		"Settings": {
			"title": "Settings fr",
			"description": "Settings Description fr",
			"properties": {
				"width": "long fr",
				"height": "not long fr"
			}
		}
	}
}
