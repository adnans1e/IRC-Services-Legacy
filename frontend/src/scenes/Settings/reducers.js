import actions from './actions';

const initialState = {
	logoutTimeout: 10,
};

const settingsReducers = (state = initialState, action) => {
	switch (action.type) {
		case actions.types.setLogoutTimeout:
			console.log('settingsReducers::setLogoutTimeout');
			return { ...state, logoutTimeout: action.payload };
		default:
			return state;
	}
};

export default settingsReducers;
