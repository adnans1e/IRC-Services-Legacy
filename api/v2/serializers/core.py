import json

from django.http import Http404
from django.utils.encoding import force_text
from django.utils.http import urlsafe_base64_decode
from rest_framework.generics import get_object_or_404
from django.contrib.auth.models import Group
from django.contrib.auth import authenticate, get_user_model

import regions.models
from api.utils import generate_translated_fields
from collections import OrderedDict
from django.conf import settings
from email_user.models import EmailUser, token_generator
from regions.models import GeographicRegion
from rest_framework import exceptions, serializers
from services.models import Service, Provider, ServiceArea
from . import apps as apps_serializers
from . import services as sevices_serializers

CAN_EDIT_STATUSES = [Service.STATUS_DRAFT,
                     Service.STATUS_CURRENT, Service.STATUS_REJECTED]
DRFValidationError = exceptions.ValidationError


class GroupSerializer(serializers.HyperlinkedModelSerializer):
    class Meta:
        model = Group
        fields = ('url', 'id', 'name')


class APILoginSerializer(serializers.Serializer):
    """
    Serializer for our "login" API.
    Both validates the call parameters and authenticates
    the user, returning the user in the validated_data
    if successful.

    Adapted from authtoken/serializers.py for our email-based user model
    """
    email = serializers.EmailField()
    password = serializers.CharField()

    def validate(self, attrs):
        attrs = super().validate(attrs)
        email = attrs.get('email')
        password = attrs.get('password')
        user = authenticate(email=email, password=password)

        if user:
            if not user.is_active:
                msg = _('User account is disabled.')
                raise exceptions.ValidationError(msg)
        else:
            msg = _('Unable to log in with provided credentials.')
            raise exceptions.ValidationError(msg)

        attrs['user'] = user
        return attrs


class GroupSerializer(serializers.HyperlinkedModelSerializer):
    class Meta:
        model = Group
        fields = ('url', 'id', 'name')


class RequireOneTranslationMixin(object):
    """Validate that for each set of fields with prefix
    in `Meta.required_translated_fields` and ending in _en, _ar, _fr,
    that at least one value is provided."""

    # Override run_validation so we can get in at the beginning
    # of validation for a call and add our own errors to those
    # the other validations find.
    def run_validation(self, data=serializers.empty):
        # data is a dictionary
        errs = defaultdict(list)
        for field in self.Meta.required_translated_fields:
            if not any(data.get(key, False) for key in generate_translated_fields(field, False)):
                errs[field].append(_('This field is required.'))
        try:
            validated_data = super().run_validation(data)
        except (exceptions.ValidationError, DjangoValidationError) as exc:
            errs.update(serializers.get_validation_error_detail(exc))
        if errs:
            raise exceptions.ValidationError(errs)
        return validated_data


class ServiceAreaSerializer(RequireOneTranslationMixin,
                            serializers.HyperlinkedModelSerializer):
    class Meta:
        model = ServiceArea
        fields = tuple(
            [
                'id',
                'parent',
                'url'
            ] + generate_translated_fields('name')
        )
        required_translated_fields = ['name']


class UserSerializer(serializers.ModelSerializer):
    managed_providers = sevices_serializers.ProviderSerializer(
        many=True, read_only=True)

    class Meta:
        model = EmailUser
        fields = ('id', 'email', 'groups', 'name', 'surname', 'is_active', 'is_staff', 'language', 'region', 'is_superuser', 'phone_number', 'title',
                  'position', 'providers', 'managed_providers')


class UserWithGroupSerializer(serializers.ModelSerializer):
    isStaff = serializers.BooleanField(source="is_staff")
    isSuperuser = serializers.BooleanField(source="is_superuser")
    groups = GroupSerializer(many=True)
    providers = sevices_serializers.ProviderSerializer(many=True,)
    managed_providers = sevices_serializers.ProviderSerializer(many=True,)

    def validate(self, attrs):
        print(self)

        return attrs

    class Meta:
        model = EmailUser
        fields = ('id', 'email', 'groups', 'name', 'surname', 'is_active', 'is_staff', 'is_superuser', 'language',  'phone_number', 'title',
                  'position', 'providers', 'isStaff', 'isSuperuser', 'region', 'managed_providers')


class UserAvatarSerializer(serializers.ModelSerializer):
    avatar = serializers.ImageField(allow_null=True)

    def validate(self, attrs):
        avatar = attrs['avatar']
        if avatar and avatar._size >= settings.MAX_UPLOAD_SIZE:
            raise DRFValidationError('File is too large. Max 2.5 MB.')
        return attrs

    class Meta:
        model = EmailUser
        fields = ('avatar',)


class EmailSerializer(serializers.Serializer):
    email = serializers.EmailField()

    def validate(self, attrs):
        try:
            get_object_or_404(EmailUser, email=attrs['email'])
            return attrs
        except Http404:
            raise DRFValidationError(
                'The e-mail address is not assigned to any user account.')


class SecurePasswordCredentialsSerializer(serializers.Serializer):
    uidb64 = serializers.RegexField(regex='[0-9A-Za-z_\-]+')
    token = serializers.RegexField(regex='[0-9A-Za-z]{1,13}-[0-9A-Za-z]{1,20}')

    def validate(self, attrs):
        token = attrs['token']
        try:
            uid = force_text(urlsafe_base64_decode(attrs['uidb64']))
            user = EmailUser.objects.get(pk=uid)
        except (TypeError, ValueError, OverflowError, EmailUser.DoesNotExist):
            raise DRFValidationError('Bad reset link.')

        if user is not None and token_generator.check_token(user, token):
            attrs['user_id'] = user.id
            return attrs
        raise DRFValidationError('Bad reset link.')


class ResetUserPasswordSerializer(serializers.Serializer):
    id = serializers.IntegerField()
    new_password1 = serializers.CharField()
    new_password2 = serializers.CharField()
    user = None

    def validate(self, attrs):
        try:
            self.user = get_object_or_404(EmailUser, pk=attrs['id'])
            if attrs['new_password1'] != attrs['new_password2']:
                raise DRFValidationError(
                    "The two password fields didn't match.")
        except Http404:
            raise DRFValidationError('Invalid user.')

        return attrs


class ProviderSerializer(serializers.ModelSerializer):
    number_of_monthly_beneficiaries = serializers.IntegerField(
        min_value=0, max_value=1000000,
        required=False,
        allow_null=True
    )

    class Meta:
        model = Provider
        fields = tuple(
            [
                'url', 'id',
            ] +
            generate_translated_fields('name') +
            generate_translated_fields('description') +
            generate_translated_fields('focal_point_name') +
            generate_translated_fields('address') +
            [
                'type', 'phone_number', 'website',
                'facebook', 'twitter',
                'focal_point_phone_number',
                'user', 'number_of_monthly_beneficiaries', 'is_frozen'
            ]
        )
        required_translated_fields = [
            'name', 'description', 'focal_point_name', 'address']


class ServiceExcelSerializer(serializers.ModelSerializer):
    location = serializers.SerializerMethodField(read_only=True)
    region = serializers.RelatedField(read_only=True)

    def get_location(self, obj):
        return ",".join([str(obj.location.y), str(obj.location.x)]) if obj.location else ''

    FIELD_MAP = OrderedDict(
        [
            ('id', 'Identifier'),
            ('region', 'Region of Service'),
            ('location', 'Coordinates'),
            ('type', 'Type of Service'),
            ('phone_number', 'Phone Number'),
        ] +
        [("name_{}".format(k), "Name in ({})".format(v)) for k, v in settings.LANGUAGES] +
        [("description_{}".format(k), "Description in ({})".format(v)) for k, v in settings.LANGUAGES] +
        [("address_{}".format(k), "Address in ({})".format(v)) for k, v in settings.LANGUAGES] +
        [
            ('sunday_open', 'Sunday Opening Hours (00:00:00)'),
            ('sunday_close', 'Sunday Closing Hours (00:00:00)'),
            ('monday_open', 'Monday Opening Hours (00:00:00)'),
            ('monday_close', 'Monday Closing Hours (00:00:00)'),
            ('tuesday_open', 'Tuesday Opening Hours (00:00:00)'),
            ('tuesday_close', 'Tuesday Closing Hours (00:00:00)'),
            ('wednesday_open', 'Wednesday Opening Hours (00:00:00)'),
            ('wednesday_close', 'Wednesday Closing Hours (00:00:00)'),
            ('thursday_open', 'Thursday Opening Hours (00:00:00)'),
            ('thursday_close', 'Thursday Closing Hours (00:00:00)'),
            ('friday_open', 'Friday Opening Hours (00:00:00)'),
            ('friday_close', 'Friday Closing Hours (00:00:00)'),
            ('saturday_open', 'Saturday Opening Hours (00:00:00)'),
            ('saturday_close', 'Sunday Closing Hours (00:00:00)'),
        ])

    class Meta:
        model = Service
        fields = (
            [
                'id',
                'region',
                'location',
                'sunday_open',
                'sunday_close',
                'monday_open',
                'monday_close',
                'tuesday_open',
                'tuesday_close',
                'wednesday_open',
                'wednesday_close',
                'thursday_open',
                'thursday_close',
                'friday_open',
                'friday_close',
                'saturday_open',
                'saturday_close',
                'type',
                'phone_number',
            ] + generate_translated_fields('name') +
            generate_translated_fields('description') +
            generate_translated_fields('address') +
            generate_translated_fields('additional_info') +
            generate_translated_fields('languages_spoken')
        )


class GeographicRegionSerializer(serializers.ModelSerializer):
    parent__name = serializers.SerializerMethodField(read_only=True)
    centroid = serializers.SerializerMethodField(read_only=True)
    envelope = serializers.SerializerMethodField(read_only=True)

    def get_centroid(self, obj):
        return json.loads(obj.centroid.json)

    def get_envelope(self, obj):
        return json.loads(obj.geom.envelope.json)

    def get_parent__name(self, obj):
        return obj.parent.name if obj.parent else ''

    class Meta:
        model = GeographicRegion
        fields = tuple(
            ['id', 'name', 'slug', 'code', 'hidden', 'level', 'geom', 'centroid', 'envelope', 'parent',
             'parent__name', 'languages_available'] +
            generate_translated_fields('title')
        )


class GeographicRegionSerializerNoGeometry(serializers.ModelSerializer):
    parent__name = serializers.SerializerMethodField(read_only=True)
    centroid = serializers.SerializerMethodField(read_only=True)
    envelope = serializers.SerializerMethodField(read_only=True)

    def get_centroid(self, obj):
        return json.loads(obj.centroid.json)

    def get_envelope(self, obj):
        return json.loads(obj.geom.envelope.json)

    def get_parent__name(self, obj):
        return obj.parent.name if obj.parent else ''

    class Meta:
        model = GeographicRegion
        fields = tuple(
            ['id', 'name', 'slug', 'code', 'hidden', 'level', 'centroid', 'envelope', 'parent',
             'parent__name', 'languages_available'] +
            generate_translated_fields('title')
        )


class UserPermissionSerializer(serializers.ModelSerializer):
    permissions = serializers.SerializerMethodField(required=False)

    def get_permissions(self, obj):
        return obj.get_all_permissions()

    class Meta:
        model = EmailUser
        fields = ('email', 'permissions')


class APIRegisterSerializer(serializers.Serializer):
    email = serializers.EmailField()
    name = serializers.CharField()
    surname = serializers.CharField()
    title = serializers.CharField(required=False)
    position = serializers.CharField(required=False)
    phone_number = serializers.CharField(required=False)
    groups = serializers.PrimaryKeyRelatedField(
        many=True, required=False, read_only=True)

    def validate(self, attrs):
        attrs = super().validate(attrs)
        User = get_user_model()
        user = User.objects.filter(email=attrs.get('email'))
        if user:
            raise exceptions.ValidationError(
                {'email': 'User with this email already exists'})
        return attrs
