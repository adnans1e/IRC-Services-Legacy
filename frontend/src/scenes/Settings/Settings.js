import React, { useState, useEffect } from 'react';
import { useTranslation } from "react-i18next";
import i18n from '../../shared/i18n';
import languages from './languages.json';
import ListView from './views/ListView/ListView';
import { textFilter } from 'react-bootstrap-table2-filter';
import './Settings.scss';
import api from './api';

const NS = 'Settings';

const Settings = props => {
	i18n.customLoad(languages, NS);
	const { t } = useTranslation(NS);

	const [data, setData] = useState([]);

	useEffect(() => {
		(async function fetchData() {
			const response = await api.settings.getAll();
			setData(response);
			console.log(response);
		})();
	}, []);

	return (
		<div className={NS}>
			Settings
		</div>
	)
}

export default Settings;
