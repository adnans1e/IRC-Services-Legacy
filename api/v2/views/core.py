import logging

from django.conf import settings
from django.contrib.auth.models import Group
from django.contrib.sites.shortcuts import get_current_site

from api.serializers import APILoginSerializer, APIRegisterSerializer
from api.v2 import serializers as serializers_v2
from django.contrib.auth import get_user_model, login
from django.db.transaction import atomic
from django.urls import reverse
from django.utils.timezone import now
from rest_framework.permissions import AllowAny, IsAuthenticated, IsAuthenticatedOrReadOnly

from api.v2.serializers import UserAvatarSerializer, EmailSerializer, SecurePasswordCredentialsSerializer, \
    ResetUserPasswordSerializer, GroupSerializer
from email_user.models import EmailUser
from regions.models import GeographicRegion
from rest_framework import permissions
from rest_framework import viewsets, mixins
from rest_framework.authtoken.models import Token
from rest_framework.decorators import list_route
from rest_framework.response import Response
from .utils import IsSuperUserPermission, StandardResultsSetPagination
from ..filters import GeographicRegionFilter

logger = logging.getLogger(__name__)


class GroupViewSet(viewsets.ModelViewSet):
    """
    API endpoint that allows groups to be viewed or edited.
    """
    queryset = Group.objects.all()
    serializer_class = GroupSerializer
class UserViewSet(viewsets.ModelViewSet):
    """
    API endpoint that allows users to be viewed or edited.
    """
    queryset = EmailUser.objects.all()
    serializer_class = serializers_v2.UserSerializer
    pagination_class = StandardResultsSetPagination

    def get_serializer_class(self):
        if getattr(self, 'action') in ('retrieve', 'list'):
            return serializers_v2.UserWithGroupSerializer
        return super().get_serializer_class()

    def get_permissions(self):
        if getattr(self, 'action') in ('retrieve', 'me', 'logout', 'partial_update', 'update'):
            self.permission_classes = [permissions.IsAuthenticated]
        elif getattr(self, 'action') in ('login', 'reset_password', 'reset_password_confirm',
                                         'reset_password_done'):
            self.permission_classes = [permissions.AllowAny]
        else:
            self.permission_classes = [IsSuperUserPermission]

        return super(UserViewSet, self).get_permissions()

    @list_route()
    def me(self, request):
        serializer_class = self.get_serializer_class()
        instance = request.user

        return Response(serializer_class(instance).data)

    @list_route(methods=['POST'])
    def login(self, request):
        serializer = APILoginSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        user = serializer.validated_data['user']
        token, created = Token.objects.get_or_create(user=user)
        user.last_login = now()
        user.save(update_fields=['last_login'])
        login(request, user)

        response = {'token': token.key,
                    'language': user.language,
                    'email': user.email,
                    'isStaff': user.is_staff,
                    'isSuperuser': user.is_superuser,
                    'name': user.name,
                    'surname': user.surname,
                    'id': user.id,
                    'title': user.title,
                    'position': user.position,
                    'phone_number': user.phone_number
                    }

        if user.avatar:
            response.update({'avatar': settings.MEDIA_URL + str(user.avatar)})

        return Response(response)

    @list_route(methods=['POST'])
    def register(self, request):
        serializer = APIRegisterSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        with atomic():
            kwargs = dict()
            if request.data.get('title', None):
                kwargs['title'] = request.data['title']
            if request.data.get('position', None):
                kwargs['position'] = request.data['position']
            if request.data.get('phone_number', None):
                kwargs['phone_number'] = request.data['phone_number']
            user = get_user_model().objects.create_user(
                name=request.data['name'],
                surname=request.data['surname'],
                email=request.data['email'],
                password=request.data['password'],
                is_active=False,
                **kwargs
            )
            if request.data.get('groups', None):
                user.groups = [Group.objects.get(
                    name=group['name']) for group in request.data['groups']]
                user.save()
            token, created = Token.objects.get_or_create(user=user)
            activation_url = request.build_absolute_uri(
                reverse('api-activate')) + '?activation_key='
            user.send_activation_email(request, activation_url)

        return Response({})

    @list_route(methods=['GET'])
    def logout(self, request):
        user = request.user
        token, created = Token.objects.get_or_create(user=user)
        token.delete()

        return Response({})

    @list_route(methods=['POST'])
    def reset_password(self, request):
        serializer = EmailSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        email = request.data['email']
        user = get_user_model().objects.get(email=email)
        site = get_current_site(request)
        base_url = request.data['base_url']
        user.send_password_reset_email(base_url, site)
        return Response({})

    @list_route(methods=['POST'])
    def reset_password_confirm(self, request):
        serializer = SecurePasswordCredentialsSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        return Response({'userId': serializer.validated_data['user_id']})

    @list_route(methods=['POST'])
    def reset_password_done(self, request):
        serializer = ResetUserPasswordSerializer(data=request.data)
        serializer.is_valid(raise_exception=True)
        user = EmailUser.objects.get(pk=serializer.data['id'])
        user.set_password(serializer.data['new_password1'])
        user.save()
        return Response({})

    def partial_update(self, request, *args, **kwargs):
        partial = kwargs.pop('partial', True)
        instance = self.get_object()
        if request.user.is_superuser or instance == request.user:
            serializer = UserAvatarSerializer(
                instance, data=request.data, partial=partial)
            serializer.is_valid(raise_exception=True)
            self.perform_update(serializer)
            return Response({'status': 204, 'avatar': settings.MEDIA_URL + str(instance.avatar.name)})
        return Response(status=403)

    def update(self, request, *args, **kwargs):
        # Don't allow users to change their email
        partial = kwargs.pop('partial', False)
        instance = self.get_object()
        if request.user.is_superuser or instance == request.user:
            serializer = self.get_serializer(
                instance, data=request.data, partial=partial)
            serializer.is_valid(raise_exception=True)
            self.perform_update(serializer)
            return Response(serializer.data)
        return Response(status=403)

    def retrieve(self, request, *args, **kwargs):
        instance = self.get_object()
        if request.user.is_superuser or instance == request.user:
            serializer = self.get_serializer(instance)
            user = serializer.data
            return Response({'email': user['email'],
                             'isStaff': user['is_staff'],
                             'isSuperuser': user['is_superuser'],
                             'name': user['name'],
                             'surname': user['surname'],
                             'id': user['id'],
                             'title': user['title'],
                             'position': user['position'],
                             'groups': user['groups'],
                             'phone_number': user['phone_number']})
        return Response(status=403)


class GeographicRegionViewSet(viewsets.ModelViewSet):
    permission_classes = [permissions.DjangoModelPermissionsOrAnonReadOnly]
    queryset = GeographicRegion.objects.select_related('parent').all()
    serializer_class = serializers_v2.GeographicRegionSerializer
    pagination_class = StandardResultsSetPagination
    filter_class = GeographicRegionFilter

    def get_serializer_class(self):
        if 'exclude_geometry' in self.request.GET:
            return serializers_v2.GeographicRegionSerializerNoGeometry
        else:
            return serializers_v2.GeographicRegionSerializer


class UserPermissionViewSet(viewsets.ReadOnlyModelViewSet):
    permission_classes = [IsAuthenticated]
    queryset = EmailUser.objects.all()
    serializer_class = serializers_v2.UserPermissionSerializer
    http_method_names = ['get']

    def list(self, request, *args, **kwargs):
        queryset = EmailUser.objects.get(pk=request.user.pk)
        serializer = self.get_serializer(queryset, many=False)
        return Response(serializer.data)
