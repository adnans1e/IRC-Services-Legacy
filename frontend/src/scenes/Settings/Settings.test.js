import React, { Suspense } from "react";
import { shallow, mount, render } from "enzyme";
import { Provider } from 'react-redux';
import  Settings from "./Settings";
import { prop } from '@uirouter/core';
//import store from '../.././shared/store';
import configureStore from 'redux-mock-store';
import { create } from "react-test-renderer";
//import { render } from '@testing-library/react'
import expectationResultFactory from "jest-jasmine2/build/expectationResultFactory";
import { findByTestAtrrr, testStore } from '../../.././Utils'
import { isRegExp } from "util";
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';

const settingsComponent = shallow(<Suspense fallback=''><Settings /></Suspense>);
console.log(settingsComponent.debug());

describe('Setting tests: ', () => {
    it('Setting component renders ok', () => {
        const tree = renderer.create(<Suspense fallback=''><Settings /></Suspense>).toJSON();
        expect(tree).toMatchSnapshot();
    }) 
  })