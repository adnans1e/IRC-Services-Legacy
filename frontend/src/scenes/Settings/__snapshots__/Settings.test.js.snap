// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting tests:  Setting component renders ok 1`] = `""`;
